cleaner.go

```diff
@@ -547,3 +548,38 @@ func (c *Cleaner) RemoveCRDs() error {
 
 	return nil
 }
+
+func (c *Cleaner) waitForPodsGone(dep unstructured.Unstructured) error {
+	path := []string{"spec", "selector", "matchLabels"}
+	ls, found, err := unstructured.NestedStringMap(dep.Object, path...)
+	if err != nil {
+		msg := fmt.Sprintf("unstructured dep %v/%v failed to find selector %v: %v", dep.GetNamespace(), dep.GetName(), path, err)
+		return gerr.Wrap(err, msg)
+	}
+	if !found {
+		return fmt.Errorf("unstructured dep %v/%v missing selector %v", dep.GetNamespace(), dep.GetName(), path)
+	}
+	cnd := func() (bool, error) {
+		pods := &v1.PodList{}
+		opts := []client.ListOption{client.InNamespace(dep.GetNamespace()), client.MatchingLabels(ls)}
+		if err := c.k8sCli.List(context.Background(), pods, opts...); err != nil {
+			return false, err
+		}
+		return len(pods.Items) == 0, nil
+	}
+	return wait.PollImmediate(5*time.Second, 10*time.Minute, cnd)
+}
+
+func (c *Cleaner) ensureServiceCatalogNotRunning() error {
+	done := make(chan error, 2)
+	go func() {
+		done <- c.waitForPodsGone(serviceCatalogCatalogControllerManagerUnstructuredDeployment)
+	}()
+	go func() {
+		done <- c.waitForPodsGone(helmBrokerUnstructuredDeployment)
+	}()
+	var errs []error
+	errs = append(errs, <-done)
+	errs = append(errs, <-done)
+	return errors.NewAggregate(errs)
```

Comment:
fwiw, the `NewAggregate` filters out `nil` errors, if the slice then is empty, it returns `nil` as well 

https://github.com/kubernetes/apimachinery/blob/df993592a122931b8aac4db57689e09458a2332c/pkg/util/errors/errors.go#L52-L59